sherpa/astro/io/dummy_backend.py

```diff
@@ -19,42 +19,766 @@
+        The name of the "block" (HDU) to read the column data (useful
+        for data structures containing multiple blocks/HDUs) or None.
+        Names are compared using a case insensitive match.
+    hdrkeys: sequence of str or None, optional
+        If set, the table structure must contain these keys, and the
+        values are returned.  Names are compared using a case
+        insensitive match.
+
+    Returns
+    -------
+    names, data, filename, hdr
+        The column names, as a list of strings, and the data as
+        a list of NumPy arrays (matching the order and length of
+        the names array). The filename is the name of the file (a
+        string) and hdr is a dictionary with the requested keywords
+        (when hdrkeys is `None` this dictionary will be empty).
+
+    Raises
+    ------
+    sherpa.utils.err.IOErr
+        The arg argument is invalid, or a required column or keyword
+        is missing.
+
+    """
+    raise NotImplementedError('No usable I/O backend was imported.')
+
+
+def get_header_data(arg,
+                    blockname: Optional[str] = None,
+                    hdrkeys: Optional[NamesType] = None
+                    ) -> HdrType:
+    """Read the metadata.
+
+    Parameters
+    ----------
+    arg
+        The data to read the header values from. This depends on the
+        backend but is expected to be a file name or a data structure
+        supported by the backend.
+    blockname: str or None, optional
+        The name of the "block" (HDU) to read the data (useful
+        for data structures containing multiple blocks/HDUs) or None.
+        Names are compared using a case insensitive match.
+    hdrkeys: sequence of str or None, optional
+        If set, the table structure must contain these keys, and the
+        values are returned. Names are compared using a case
+        insensitive match.
+
+    Returns
+    -------
+    hdr: dict
+        A dictionary with the keyword data (only the name and values
+        are returned, any sort of metadata, such as comments or units,
+        are not returned).
+
+    Raises
+    ------
+    sherpa.utils.err.IOErr
+        The arg argument is invalid or a keyword is missing.
+
+    """
+    raise NotImplementedError('No usable I/O backend was imported.')
+
+
+def get_image_data(arg,
+                   make_copy: bool = True,
+                   fix_type: bool = True
+                   ) -> tuple[DataType, str]:
+    """Read image data.
+
+    Parameters
+    ----------
+    arg
+        The data to read the header values from. This depends on the
+        backend but is expected to be a file name or a data structure
+        supported by the backend.
+    make_copy: bool, optional
+        If set then the returned NumPy arrays are explictly copied,
+        rather than using a reference from the data structure
+        created by the backend. Backends are not required to
+        honor this setting. The default is `True`.
```

Comment:
My feeling is that python documentation guidelines have changed over time, in part because of the increasing prevalence of  language-server-esque capabilities.

In general I'm okay with transitioning to removing the "The default is blah" statements, but I think it's a case-by-case basis (e.g. for the UI code I'd prefer to keep it because it may get used when we create the CIAO ahelp files where the documentation rules are different).

Reply:
I created #2088 to track this.